CFC_DataCollector/recommender/common.py

```diff
@@ -0,0 +1,162 @@
+	to_insert['single_mode'] = str(single_mode)
+	to_insert['legs'] = [ ]
+	for leg in trip.legs:
+		l = { }
+		l['mode'] = leg.mode
+		l['cost'] = leg.cost
+		l['duration'] = leg.duration
+		l['distance'] = distance
+		l['start_point'] = {'type' : 'Point', 'coordinates' : leg.starting_point}
+		l['end_point'] = {'type' : 'Point', 'coordinates' : leg.ending_point}
+		to_insert['legs'].append(l)
+	if is_alternate:
+		t = db.find_one({'trip_id' : trip.parent_tid})
+		try:
+			t['alternate_trips'].append(to_insert)
+		except:
+			t['alternate_trips'] = []
+			t['alternate_trips'].append(to_insert)
+	else:
+		db.insert(to_insert)
+
+
+def trip_to_json(trip):
+	to_insert = { }
+	to_insert['_id'] = trip._id
+	#to_insert['distance'] = trip.distance
+	to_insert['section_start_datetime'] = trip.start_time
+	to_insert['section_end_datetime'] = trip.end_time
+	to_insert['section_start_point'] = {'type' : 'Point', 'coordinates' : trip.start_point}
+	to_insert['section_end_point'] = {'type' : 'Point', 'coordinates' : trip.end_point}
+	to_insert['cost'] = trip.cost
+	to_insert['single_mode'] = str(single_mode)
+	to_insert['legs'] = [ ]
+	for leg in trip.legs:
+		l = { }
+		l['mode'] = leg.mode
+		l['cost'] = leg.cost
+		l['duration'] = leg.duration
+		l['distance'] = distance
+		l['start_point'] = {'type' : 'Point', 'coordinates' : leg.starting_point}
+		l['end_point'] = {'type' : 'Point', 'coordinates' : leg.ending_point}
+		to_insert['legs'].append(l)
+	final = {tripObj._id : to_insert}
+	return final
+
+def create_trip_id():
+	return random.randint(100, 999)
+
+def string_start_time_to_datetime(start_time_string):
+	year = int(start_time_string[:4])
+	month = int(start_time_string[4:6])
+	day = int(start_time_string[6:8])
+	hour = int(start_time_string[9:11])
+	minute = int(start_time_string[11:13])
+	second = int(start_time_string[13:15])
+	return datetime(year, month, day, hour, minute, second)
+
+def datetime_to_string(datetime_obj):
+	temp = str(datetime_obj)
+	return "%s%s%sT%s%s%s-0700" % (temp[:4], temp[5:7], temp[8:10], temp[11:13], temp[14:16], temp[17:19])
+
+
+
+def find_perturbed_trips(trip, delta=2):
+    to_return = [ ]
+    time_delta = timedelta(minutes=delta)
+    fifteen_min = timedelta(minutes=15)
+    original_delta = string_start_time_to_datetime(trip.end_time) - string_start_time_to_datetime(trip.start_time) 
+    start = string_start_time_to_datetime(trip.start_time) - fifteen_min
+    end = string_start_time_to_datetime(trip.end_time) + fifteen_min
+    time = start
+    while time < end:
+    	_id = str(create_trip_id()) + str(trip._id) 
+    	json_str = {}
+    	json_str['section_start_time'] = datetime_to_string(time)
+    	json_str['section_end_time'] = datetime_to_string(time + original_delta)   ##Asuming the perturbed trip takes as long as the original trip
+    	json_str['_id'] = _id 
+    	json_str['mode'] = trip.single_mode
+    	json_str['track_points'] = None
+        new_trip = E_Mission_Trip(json_str)
```

Comment:
@joshzarrabi, @sdsingh E_Mission_Trip classes are for real trips. Perturbed trips should be their own subclass of trip.